src/components/GraphiQL.js

```diff
@@ -297,6 +308,13 @@ export class GraphiQL extends React.Component {
               </div>
             </div>
             <div className="resultWrap">
+              {(() => {
+                return this.state.showSpinner && (
+                  <div className="spinner-container">
+                    <div className="spinner" />
+                  </div>
+                );
+              })()}
               <ResultViewer ref="result" value={this.state.response} />
```

Comment:
Perhaps the ResultViewer should be grayed out or something while another query is underway and the spinner is placed over it.